src/types.ts

```diff
@@ -127,38 +128,50 @@ export enum CertificateType {
   POOL_RETIREMENT = 4,
   GENESIS_KEY_DELEGATION = 5,
   MOVE_INSTANTANEOUS_REWARDS_CERT = 6,
+  STAKE_REG = 7,
+  STAKE_UNREG = 8,
+  VOTE_DELEG = 9,
+  STAKE_VOTE_DELEG = 10,
+  STAKE_REG_DELEG = 11,
+  VOTE_REG_DELEG = 12,
+  STAKE_VOTE_REG_DELEG = 13,
+  AUTH_COMMITTEE_HOT = 14,
+  RESIGN_COMMITTEE_COLD = 15,
+  DREP_REG = 16,
+  DREP_UNREG = 17,
+  DREP_UPDATE = 18,
```

Comment:
Regarding the certificate naming, what do you think about the following:

```suggestion
  STAKE_REGISTRATION_CONWAY = 7,
  STAKE_UNREGISTRATION_CONWAY = 8,
  VOTE_DELEGATION = 9,
  STAKE_AND_VOTE_DELEGATION = 10,
  STAKE_REGISTRATION_AND_DELEGATION = 11,
  STAKE_REGISTRATION_AND_VOTE_DELEGATION = 12,
  STAKE_REGISTRATION_AND_STAKE_AND_VOTE_DELEGATION = 13,
  AUTHORIZE_COMMITTEE_HOT = 14,
  RESIGN_COMMITTEE_COLD = 15,
  DREP_REGISTRATION = 16,
  DREP_UNREGISTRATION = 17,
  DREP_UPDATE = 18,
```

The names are rather long but at least it would be clear what's happening.

Reply:
I'd also consider using **de**registration instead of **un**registration since there already is a STAKE_**DE**REGISTRATION_CERTIFICATE.

Reply:
How would you suggest to rename `StakeVoteRegDelegCertificate` ?
`StakeRegistrationAndStakeAndVoteDelegationCertificate`? Sounds a bit too much :) Perhaps the second And can be left out? (Having two `and`s with different levels is better avoided.)

They shouldn't have included those combinations at all...